backend/src/main.rs

```diff
@@ -35,7 +38,23 @@ async fn main() -> Result<()> {
     }
 
     // Parse CLI args.
-    let args = Args::from_args();
+    // This is a bit roundabout because we want to override the version
+    // using some runtime code.
+    let args = Args::from_clap(
+        &Args::clap()
+            .version(&*format!(
+                "{} ({}{}), built {}",
+                build_info::PKG_VERSION,
+                build_info::GIT_COMMIT_HASH.unwrap(),
+                if let Some(true) = build_info::GIT_DIRTY {
+                    ", dirty"
+                } else {
+                    ""
+                },
+                build_info::BUILT_TIME_UTC,
+            ))
+            .get_matches(),
+    );
```

Comment:
I would prefer to extract the `format!()` thingy and create a new variable `version`. This avoids this very complex expression and improves readability IMO.

EDIT: ah, I see, you change it next commit. This commit could have had the `version` function already but oh well, doesn't matter I guess :D

Reply:
I got you :ok_hand: